mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceProperties.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.app.mail.source;
+
+import java.util.Properties;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Configuration properties for the Mail Source module.
+ *
+ * @author Amol
+ */
+@ConfigurationProperties
+public class MailSourceProperties {
```

Comment:
Added expression for selector. 
I think IMAP properties works fine with `javaMailProperties`. Same for POP3

Tested with gmail and exchange server for Imap and IdleImap 
Test on another server for POP3


Used the below parameters for IMAP . 
`--protocol=imaps --mail-url=imaps://username%40gmail.com:password@imap.gmail.com:993/INBOX --mark-as-read=true --delete=false --java-mail-properties=mail.imap.socketFactory.class=javax.net.ssl.SSLSocketFactory\nmail.imap.socketFactory.fallback=false\nmail.store.protocol=imaps\nmail.debug=true --spring.cloud.stream.bindings.output.destination=mail1`